### PR TITLE
Fix babel-ts to emit a proper syntax error for `(a:b)`

### LINKED ISSUE #8041: babel-ts crashes on `(a:b)`
**Prettier 2.0.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAKAhkgRgShAGhAgAcYBLaAZ2VHQCc6IB3ABXoWpXQBsn0BPaoSx10YANZwYAZWJiyUAObIYdAK5xCACxgBbbgHUtZeJTlg40jibIA3E-2ThKQkAspw6MFqMW70yABmPB6EAFaUAB4AQqISUtLounAAMgpwQSGaIBGR0gqK3HAAimoQ8JncoSBydB50TljoWHDcALQwrsR0CjAGZAAmMFrIABwADITdEB4GosRO3XD1thmEAI5l8D4knCDolG1QcHADpwQgdHCbZFc+6H4BSMFV2R66ZCrqbwVFpeUZZ5ZQgwZr9IYjJAAJhBojI3AKAGEILp-E5lgBWC5qDwAFWanBe1VsGgAklAzrBpGAeqQAIIU6QwfhFSoeAC+7KAA)
```sh
--parser babel-ts
```

**Input:**
```tsx
(a:b)
```

**Output:**
```tsx
printPathNoParens@https://prettier.io/lib/standalone.js:28544:15
genericPrint$3@https://prettier.io/lib/standalone.js:26180:49
callPluginPrintFunction@https://prettier.io/lib/standalone.js:16919:20
printGenerically/res<@https://prettier.io/lib/standalone.js:16869:72
printComments@https://prettier.io/lib/standalone.js:16542:26
printGenerically@https://prettier.io/lib/standalone.js:16869:24
call@https://prettier.io/lib/standalone.js:16662:30
printPathNoParens@https://prettier.io/lib/standalone.js:26439:31
genericPrint$3@https://prettier.io/lib/standalone.js:26180:49
callPluginPrintFunction@https://prettier.io/lib/standalone.js:16919:20
printGenerically/res<@https://prettier.io/lib/standalone.js:16869:72
printComments@https://prettier.io/lib/standalone.js:16542:26
printGenerically@https://prettier.io/lib/standalone.js:16869:24
printStatementSequence/<@https://prettier.io/lib/standalone.js:28568:32
map@https://prettier.io/lib/standalone.js:16728:23
printStatementSequence@https://prettier.io/lib/standalone.js:28552:10
printPathNoParens/<@https://prettier.io/lib/standalone.js:26406:18
call@https://prettier.io/lib/standalone.js:16662:30
printPathNoParens@https://prettier.io/lib/standalone.js:26405:25
genericPrint$3@https://prettier.io/lib/standalone.js:26180:49
callPluginPrintFunction@https://prettier.io/lib/standalone.js:16919:20
printGenerically/res<@https://prettier.io/lib/standalone.js:16869:72
printComments@https://prettier.io/lib/standalone.js:16542:26
printGenerically@https://prettier.io/lib/standalone.js:16869:24
call@https://prettier.io/lib/standalone.js:16662:30
printPathNoParens@https://prettier.io/lib/standalone.js:26390:25
genericPrint$3@https://prettier.io/lib/standalone.js:26180:49
callPluginPrintFunction@https://prettier.io/lib/standalone.js:16919:20
printGenerically/res<@https://prettier.io/lib/standalone.js:16869:72
printComments@https://prettier.io/lib/standalone.js:16542:26
printGenerically@https://prettier.io/lib/standalone.js:16869:24
printAstToDoc@https://prettier.io/lib/standalone.js:16879:31
coreFormat@https://prettier.io/lib/standalone.js:17147:25
format@https://prettier.io/lib/standalone.js:17367:77
formatWithCursor@https://prettier.io/lib/standalone.js:17383:14
withPlugins/<@https://prettier.io/lib/standalone.js:32885:14
format@https://prettier.io/lib/standalone.js:32894:14
formatCode@https://prettier.io/worker.js:233:21
handleMessage@https://prettier.io/worker.js:184:18
self.onmessage@https://prettier.io/worker.js:154:14
EventHandlerNonNull*@https://prettier.io/worker.js:151:1

```

**Expected behavior:**
Print a regular parse error message.

### PR DESCRIPTION
**Pull request by thorn0**
Show a proper syntax error instead.

Fixes #8041 

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Is it possible to access [this `TSErrors`](https://github.com/babel/babel/blob/31b361b736ee519180f9ea341e2a3e4e710d2ef4/packages/babel-parser/src/plugins/typescript/index.js#L62)?

**Comment by thorn0** [comment]
No, they aren't exported.

**Comment by fisker** [comment]
```js
function getTSUnexpectedTypeAnnotation() {
	try{
	  parser('(a:b)');
	} catch({message}){
	  return message;
	}
}
```

Too hacky?  😄

**Comment by thorn0** [comment]
We have it under control. The test will fail if the message suddenly changes.

File: src/language-js/parser-babel.js [hunk 1]
```diff
@@ -129,6 +129,20 @@ function tryCombinations(fn, combinations) {
   throw error;
 }
 
+function rethrowSomeRecoveredErrors(ast) {
+  if (ast.errors) {
+    for (const error of ast.errors) {
+      if (
+        typeof error.message === "string" &&
+        error.message.startsWith("Did not expect a type annotation here.")
```
Comment (fisker) [inline src/language-js/parser-babel.js:137]:
```suggestion
        error.message === "Did not expect a type annotation here."
```

We have tests, let's use simple test?

Comment (thorn0) [reply src/language-js/parser-babel.js:137]: The full message also includes `(${loc.line}:${loc.column})`.

Comment (fisker) [reply src/language-js/parser-babel.js:137]: Okay.

**Pull request merged**